src/custom_nodes/model_zoo_intel_object_detection/model_zoo_intel_object_detection.cpp

```diff
@@ -79,64 +120,170 @@ bool copy_images_into_output(struct CustomNodeTensor* output, const std::vector<
     output->dims[0] = outputBatch;
     output->dims[1] = 1;
     output->dims[2] = 4;
     output->precision = FP32;
     return true;
 }
 
-bool copy_confidences_into_output(struct CustomNodeTensor* output, const std::vector<float>& confidences) {
+bool copy_confidences_into_output(struct CustomNodeTensor* output, const std::vector<float>& confidences, void* customNodeLibraryInternalManager) {
     const uint64_t outputBatch = confidences.size();
     uint64_t byteSize = sizeof(float) * outputBatch;
+    CustomNodeLibraryInternalManager* internalManager = static_cast<CustomNodeLibraryInternalManager*>(customNodeLibraryInternalManager);
 
-    float* buffer = (float*)malloc(byteSize);
-    NODE_ASSERT(buffer != nullptr, "malloc has failed");
+    float* buffer = nullptr;
+    if (!get_preallocated_buffer<float>(internalManager, &buffer, OUTPUT_CONFIDENCES_TENSOR_NAME, byteSize))
+        return false;
     std::memcpy(buffer, confidences.data(), byteSize);
-
     output->data = reinterpret_cast<uint8_t*>(buffer);
     output->dataBytes = byteSize;
+
+    u_int64_t* dims = nullptr;
+    if (!get_preallocated_buffer<u_int64_t>(internalManager, &dims, OUTPUT_CONFIDENCES_DIMS_NAME, 3 * sizeof(uint64_t))) {
+        release(buffer, internalManager);
+        return false;
+    }
     output->dimsCount = 3;
-    output->dims = (uint64_t*)malloc(output->dimsCount * sizeof(uint64_t));
-    NODE_ASSERT(output->dims != nullptr, "malloc has failed");
+    output->dims = dims;
     output->dims[0] = outputBatch;
     output->dims[1] = 1;
     output->dims[2] = 1;
     output->precision = FP32;
     return true;
 }
 
-void cleanup(CustomNodeTensor& tensor) {
-    free(tensor.data);
-    free(tensor.dims);
+int initializeInternalManager(void** customNodeLibraryInternalManager, const struct CustomNodeParam* params, int paramsCount) {
+    // creating InternalManager instance
+    CustomNodeLibraryInternalManager* internalManager = new CustomNodeLibraryInternalManager();
+    if (internalManager == nullptr)
+        return 1;
+
+    // reading parameters to determine size of pre-allocated buffers
+    uint64_t maxOutputBatch = get_int_parameter("max_output_batch", params, paramsCount, 100);
+    NODE_ASSERT(maxOutputBatch > 0, "max output batch must be larger than 0");
+    bool convertToGrayScale = get_string_parameter("convert_to_gray_scale", params, paramsCount) == "true";
+    int targetImageHeight = get_int_parameter("target_image_height", params, paramsCount, -1);
+    int targetImageWidth = get_int_parameter("target_image_width", params, paramsCount, -1);
+    NODE_ASSERT(targetImageHeight > 0, "target image height must be larger than 0");
+    NODE_ASSERT(targetImageWidth > 0, "target image width must be larger than 0");
+
+    // creating BuffersQueues for output: images
+    int channels = convertToGrayScale ? 1 : 3;
+    uint64_t imagesByteSize = sizeof(float) * targetImageHeight * targetImageWidth * channels * maxOutputBatch;
```

Comment:
This is max output batch. In case of `demultiply_count=0` (which happens very often, even in our documented examples return arbitrary number of batches (max `maxOutputBatch`).

Reply:
What if we have `max_output_batch=100` but execution generates only bs=5?

Reply:
In initialize we are creating BuffersQueues without any knowledge about bs. Therefore we are preparing buffers to handle maximum expected size. 

In conclusion, during initialize we are allocating buffers with max size, but during copy_* operations only needed amount will be used, but whole stream is locked by that client. So with max_output_batch=100 and bs=5 only 5% of pre-allocated memory will be used and 95% will be locked.

Reply:
Would it be an issue? @atobiszei